### PR TITLE
Top read articles!

### PR DESCRIPTION
**Pull request by bgerstle**
#Phab: [T124716](https://phabricator.wikimedia.org/T124716), 

---

- [x] Get previews of (somewhat-filtered) top-read articles into feed
- [x] Remove extract from preview request
- [x] Test across multiple days
- [x] Incorporate title blacklist
- [x] Get preview namespace and filter out anything but "Main"
- [x] Add "more top read articles" :sparkles: 

image

### CODE REVIEW

**Comment by montehurd** [comment]
Didn't see title and title prefix exclusions on your checklist.

I built these exclusion lists by setting the day to yesterday, looking at the results, seeing what ones didn't make sense (i.e. xhamster porn, "-", file pages for images, etc), then going back to the day before that and looking again, and repeated this for iirc 30 days or so:

https://github.com/wikimedia/wikipedia-ios/pull/388/files#diff-242c970b3f03da0b80d49c34a5477b41R43

**Comment by montehurd** [comment]
@bgerstle 
> can haz more top read articles plz kthxbai

lol

File: Wikipedia/Code/MWKSearchResult.m [hunk 1]
```diff
@@ -90,11 +96,8 @@ + (NSValueTransformer*)isDisambiguationJSONTransformer {
         }
         // HAX: occasionally the search api doesn't report back "disambiguation" page term ( T121288 ),
         // so double-check wiki data description for "disambiguation page" string.
-        NSArray* description = value[@"terms.description"];
-        if (description.count && [description.firstObject containsString:@"disambiguation page"]) {
-            return @YES;
-        }
-        return @NO;
+        NSArray* descriptions = value[@"terms.description"];
+        return @(descriptions.count && [descriptions.firstObject containsString:@"disambiguation page"]);
```
Comment (montehurd) [inline Wikipedia/Code/MWKSearchResult.m]: Oops I should have done that :)

File: Wikipedia/Code/NSNumber+MWKTitleNamespace.h [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+//
+
+#import <Foundation/Foundation.h>
+
+/**
+ *  List of all built-in MediaWiki namespaces.
+ *
+ *  see https://www.mediawiki.org/wiki/Manual:Namespace#Built-in_namespaces
+ */
+typedef NS_ENUM (NSInteger, MWKTitleNamespace) {
```
Comment (montehurd) [inline Wikipedia/Code/NSNumber+MWKTitleNamespace.h]: Woohoo!

**Comment by bgerstle** [comment]
@nirzar updated copy for section header to include language:

image

**Comment by nirzar** [comment]
@bgerstle can we remove the "on" before the date. it reads oddly, Top read ON english wikipedia ON date the date can be on its own. 

and the more can be "See all top read articles"

The separator above the more is missing?

**Comment by bgerstle** [comment]
>  can we remove the "on" before the date. it

@nirzar sorry, josh actually requested "for $date". would that be better or should i revert to having the date by itself?

> The separator above the more is missing?

Screenshot artifact due to less-than-natural-size scaling of simulator:

image

> and the more can be "See all top read articles"

It could, but we're not showing "all" of them.

**Comment by montehurd** [comment]
@bgerstle 
> It could, but we're not showing "all" of them.

I don't think excluding the top 5 from the full list of top articles makes sense. It's actually confusing :(

**Comment by bgerstle** [comment]
@nirzar @jtminor i just realized: all the other "daily" sections are formatted "Tuesday, February 16". should "most read" use the same format?

image

image

Main Page too, but I'm not including it here since it's sort of phabricated

**Comment by bgerstle** [comment]
> I don't think excluding the top 5 from the full list of top articles makes sense

@montehurd Not sure what you mean, the top 5 aren't excluded. I only meant that we're not showing the full list of several hundred as returned by the pageviews API. Currently, we should the first 5, then tap "More" to reveal the longer list of 30 or so (depending on filtering, etc.)

**Comment by montehurd** [comment]
@bgerstle ah gotcha

**Comment by montehurd** [comment]
@bgerstle but to your point, "all" doesn't have to mean it's the same number the api returns. That's an arbitrary number.

**Comment by bgerstle** [comment]
@montehurd i guess "all" is actually more accurate in this case because that's "all" of them that we're showing.

**Comment by bgerstle** [comment]
@nirzar @jtminor:

image

**Comment by nirzar** [comment]
@bgerstle the separator above 4th top read article is missing 
.
.
.
just kidding :P

**Comment by bgerstle** [comment]
@nirzar :rage1:

**Comment by coreyfloyd** [comment]
:+1:

**Comment by bgerstle** [comment]
@montehurd fixed issue w/ thumbnail resolutions

**Pull request merged**